vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/get_wallet/get_wallet.cc

```diff
@@ -0,0 +1,118 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/endpoint/promotion/get_wallet/get_wallet.h"
+
+#include <utility>
+
+#include "base/json/json_reader.h"
+#include "bat/ledger/internal/endpoint/promotion/promotions_util.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "net/http/http_status_code.h"
+
+using std::placeholders::_1;
+
+namespace ledger {
+namespace endpoint {
+namespace promotion {
+
+GetWallet::GetWallet(LedgerImpl* ledger) : ledger_{ledger} {
+  DCHECK(ledger_);
+}
+
+GetWallet::~GetWallet() = default;
+
+void GetWallet::Request(GetWalletCallback callback) const {
+  auto request = type::UrlRequest::New();
+  request->url = GetUrl();
+  ledger_->LoadURL(std::move(request),
+                   std::bind(&GetWallet::OnRequest, this, _1, callback));
+}
+
+std::string GetWallet::GetUrl() const {
+  const auto rewards_wallet = ledger_->wallet()->GetWallet();
+  if (!rewards_wallet) {
+    BLOG(0, "Rewards wallet is null!");
+    return {};
+  }
+
+  return GetServerUrl("/v3/wallet/" + rewards_wallet->payment_id);
+}
+
+void GetWallet::OnRequest(const type::UrlResponse& response,
+                          GetWalletCallback callback) const {
+  ledger::LogUrlResponse(__func__, response);
+
+  type::Result result = CheckStatusCode(response.status_code);
+  if (result != type::Result::LEDGER_OK) {
+    return callback(result, std::string{}, false);
+  }
+
+  std::string custodian{};
+  bool linked{};
```

Comment:
Nit: I would say we should use `= false` (and `= ""`) just to match surrounding Chromium style code.

Reply:
Amended.